lms/djangoapps/course_api/blocks/transformers/milestones.py

```diff
@@ -35,22 +44,79 @@ def collect(cls, block_structure):
         block_structure.request_xblock_fields('is_proctored_enabled')
         block_structure.request_xblock_fields('is_practice_exam')
         block_structure.request_xblock_fields('is_timed_exam')
+        block_structure.request_xblock_fields('entrance_exam_id')
+
+    def transform(self, usage_info, block_structure):
+        """
+        Modify block structure according to the behavior of milestones and special exams.
+        """
+
+        def add_special_exam_info(block_key):
+            """
+            Adds special exam information to course blocks.
+            """
+            if self.is_special_exam(block_key, block_structure):
+
+                #
+                # call into edx_proctoring subsystem
+                # to get relevant proctoring information regarding this
+                # level of the courseware
+                #
+                # This will return None, if (user, course_id, content_id)
+                # is not applicable
+                #
+                timed_exam_attempt_context = None
+                try:
+                    timed_exam_attempt_context = get_attempt_status_summary(
+                        usage_info.user.id,
+                        unicode(block_key.course_key),
+                        unicode(block_key)
+                    )
+                except ProctoredExamNotFoundException as ex:
+                    log.exception(ex)
+
+                if timed_exam_attempt_context:
+                    # yes, user has proctoring context about
+                    # this level of the courseware
+                    # so add to the accordion data context
+                    block_structure.set_transformer_block_field(
+                        block_key,
+                        self,
+                        'special_exam',
+                        timed_exam_attempt_context,
+                    )
 
-    def transform_block_filters(self, usage_info, block_structure):
-        if usage_info.has_staff_access:
-            return [block_structure.create_universal_filter()]
+        root_key = block_structure.root_block_usage_key
+        course_key = root_key.course_key
+        user_can_skip = EntranceExamConfiguration.user_can_skip_entrance_exam(usage_info.user, course_key)
```

Comment:
@andy-armstrong I'm wondering 2 related things:

1. As I think @dianakhuang had intended, should some of this be in a separate EntranceExamTransformer?
2. If the transformer could be supplied from elsewhere, would that solve your issue of it not living in the api?
I'm not sure how much work it would be to do #2, but possible doing #1 would better enable that?